common/reactive-grpc-common/src/main/java/com/salesforce/reactivegrpc/common/AbstractStreamObserverAndPublisher.java

```diff
@@ -307,9 +286,7 @@ public void onNext(T t) {
             return;
         }
 
-        final Queue<T> q = this.queue;
-
-        while (!q.offer(t)) {
+        while (!queue.offer(t)) {
```

Comment:
Previous technique put `Queue` into the local variable, so in case of spin-lock, we will reduce a couple of redundant access to the field. In general, not sure whether previous solutions was 100% better than current

Reply:
`this.queue` is a final field. I imagine the compiler would optimize the field access.

Reply:
right